lib/agenda.js

```diff
@@ -404,32 +410,71 @@ Agenda.prototype.stop = function(cb) {
 Agenda.prototype._findAndLockNextJob = function(jobName, definition, cb) {
   var self = this,
       now = new Date(),
-      lockDeadline = new Date(Date.now().valueOf() - definition.lockLifetime);
+      lockDeadline = new Date(now.valueOf() - definition.lockLifetime);
+
+  var globalLimit = self._lockLimit ? self._lockLimit - self._lockedJobs.length : undefined;
+  if (globalLimit !== undefined) {
+    globalLimit = globalLimit <= 0 ? 0 : globalLimit;
+  }
+
+  var defLimit = definition.lockLimit ? definition.lockLimit - definition.locked : undefined;
+  if (defLimit !== undefined) {
+    defLimit = defLimit < 0 ? 0 : defLimit;
+  }
+
+  var limit = Math.min(globalLimit || 9999, defLimit || 9999);
+
+  if (limit === 0) {
+    return cb(null, null);
+  } else if (limit === undefined) {
+    limit = 0;
+  }
```

Comment:
Why is another level of lock limiting happening here? Seems redundant as it is happening [here already](https://github.com/rschmukler/agenda/blob/f25deb84acf843c7ed706aa28fc3ac9c7b4b8f5c/lib/agenda.js#L586) immediately before this method is being called.